src/serde/deserialize_program.rs

```diff
@@ -104,6 +106,29 @@ impl<'de> de::Visitor<'de> for MaybeRelocatableVisitor {
     }
 }
 
+struct ReferenceIdsVisitor;
+
+impl<'de> de::Visitor<'de> for ReferenceIdsVisitor {
+    type Value = HashMap<String, BigInt>;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("a map with string keys and integer values")
+    }
+
+    fn visit_map<A>(self, mut map: A) -> Result<Self::Value, A::Error>
+    where
+        A: MapAccess<'de>,
+    {
+        let mut data: HashMap<String, BigInt> = HashMap::new();
+
+        while let Some((key, value)) = map.next_entry::<String, usize>()? {
+            data.insert(key, BigInt::from_bytes_le(Sign::Plus, &value.to_le_bytes()));
```

Comment:
Does this only take positive numbers?

Reply:
actually no, I hardcoded that to have a quick prototype working. I will change that in a second, thanks!